src/pybind/mgr/cephadm/upgrade.py

```diff
@@ -218,7 +224,10 @@ def upgrade_ls(self, image: Optional[str], tags: bool) -> Dict:
                     continue
                 if '-' in v[2]:
                     continue
-                versions.append('.'.join(v))
+                v_major = int(v[0])
+                v_minor = int(v[1])
+                if v_major > current_major or (v_major == current_major and v_minor >= current_minor):
```

Comment:
https://pulpito.ceph.com/adking-2022-04-19_04:07:58-orch:cephadm-wip-adk-testing-2022-04-18-2113-distro-basic-smithi/6796317/
```
2022-04-19T05:29:18.936 DEBUG:teuthology.orchestra.run:got remote process result: 1
2022-04-19T05:29:18.937 ERROR:teuthology.run_tasks:Saw exception from tasks.
Traceback (most recent call last):
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_b21f9558629bbbb865321b6e62e8b2c23e4d3291/teuthology/run_tasks.py", line 91, in run_tasks
    manager = run_one_task(taskname, ctx=ctx, config=config)
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_b21f9558629bbbb865321b6e62e8b2c23e4d3291/teuthology/run_tasks.py", line 70, in run_one_task
    return task(**kwargs)
  File "/home/teuthworker/src/git.ceph.com_ceph-c_ca2dd75f248daf78c6ea1f7118a600f23cfefd9c/qa/tasks/cephadm.py", line 1151, in shell
    extra_cephadm_args=args)
  File "/home/teuthworker/src/git.ceph.com_ceph-c_ca2dd75f248daf78c6ea1f7118a600f23cfefd9c/qa/tasks/cephadm.py", line 49, in _shell
    **kwargs
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_b21f9558629bbbb865321b6e62e8b2c23e4d3291/teuthology/orchestra/remote.py", line 509, in run
    r = self._runner(client=self.ssh, name=self.shortname, **kwargs)
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_b21f9558629bbbb865321b6e62e8b2c23e4d3291/teuthology/orchestra/run.py", line 455, in run
    r.wait()
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_b21f9558629bbbb865321b6e62e8b2c23e4d3291/teuthology/orchestra/run.py", line 161, in wait
    self._raise_for_status()
  File "/home/teuthworker/src/git.ceph.com_git_teuthology_b21f9558629bbbb865321b6e62e8b2c23e4d3291/teuthology/orchestra/run.py", line 183, in _raise_for_status
    node=self.hostname, label=self.label
teuthology.exceptions.CommandFailedError: Command failed on smithi059 with status 1: "sudo /home/ubuntu/cephtest/cephadm --image docker.io/ceph/ceph:v15.2.0 shell -c /etc/ceph/ceph.conf -k /etc/ceph/ceph.client.admin.keyring --fsid 260b5c48-bf9f-11ec-8c38-001a4aab830c -- bash -c 'ceph orch upgrade ls --image quay.io/ceph/ceph | grep 16.2.0'"
```

From 
https://github.com/ceph/ceph/blob/master/qa/suites/orch/cephadm/upgrade/5-upgrade-ls.yaml
the test is expecting to see 16.2.0 and 16.2.2 in the upgrade ls output. The test needs to be modified along with this change so it still passes.